src/improvements/doc/NEW_TEMPLATE_GUIDE.md

```diff
@@ -0,0 +1,273 @@
+# New Template Creation Guide
+
+This guide explains how to create new Solidity templates for the superchain task system. Templates are the foundation for standardizing and securing task execution across different networks.
+
+## Quick Start
+
+To scaffold a new template:
+
+```bash
+cd src/improvements/
+just new template
+```
+
+This will create a new Solidity file in `src/improvements/template/` with the basic structure required.
+
+## Template Structure
+
+### 1. Required Components
+
+Every template must implement the following functions:
+
+```solidity
+// Required data structures
+struct TaskConfig {
+    // Template-specific configuration parameters
+}
+
+// Storage mapping
+mapping(uint256 => TaskConfig) public taskConfig;
+
+// Core functions
+function safeAddressString() internal pure override returns (string memory);
+function _taskStorageWrites() internal pure override returns (string[] memory);
+function _templateSetup(string memory taskConfigFilePath) internal override;
+function _build(uint256 chainId) internal override;
+function _validate(uint256 chainId) internal override;
+```
+
+The struct and mapping are optional and can be customized based on task requirements. As a general rule, task developers should think of everything as a template, hence the chainId flag to the build and validate functions.
+
+### 2. Function Implementations
+
+#### safeAddressString
+Returns the name of the multisig address from the superchain-registry:
+```solidity
+function safeAddressString() internal pure override returns (string memory) {
+    return "OptimismMultisig";
+}
+```
+
+#### _taskStorageWrites
+Lists addresses from superchain-registry whose storage will be modified:
+```solidity
+function _taskStorageWrites() internal pure override returns (string[] memory) {
+    return ["SystemConfig"];
+}
+```
+
+All addresses listed here must have their storage modified during task execution.
+
+#### _templateSetup
+Initializes the task state from config.toml:
+```solidity
+function _templateSetup(string memory taskConfigFilePath) internal override {
+    // Parse config file
+    bytes memory configBytes = vm.parseToml(vm.readFile(taskConfigFilePath));
+    
+    // Save configuration to storage
+    for (uint256 i; i < l2Chains.length; i++) {
+        uint256 chainId = l2Chains[i].chainId;
+        taskConfig[chainId] = TaskConfig({
+            // Set template-specific parameters
+        });
+    }
+}
+```
+
+#### _build
```

Comment:
I _think_ a TODO here might be useful if we need to come back to this after OPCM integration.